mpd/mpd.go

```diff
@@ -284,9 +331,36 @@ func (as *AdaptationSet) AddNewContentProtectionSchemePlayready(pro string) (*Pl
 		PRO:            Strptr(pro),
 	}
 	cp.SchemeIDURI = Strptr(CONTENT_PROTECTION_PLAYREADY_SCHEME_ID)
+
+	return cp, nil
+}
+
+// AddNewContentProtectionSchemePlayreadyWithPSSH adds a new content protection scheme for PlayReady DRM. The scheme
+// will include both ms:pro and cenc:pssh subelements
+// pro - PlayReady Object Header, as a Base64 encoded string.
+func (as *AdaptationSet) AddNewContentProtectionSchemePlayreadyWithPSSH(pro string) (*PlayreadyContentProtection, error) {
+	cp, err := newPlayreadyContentProtection(pro)
+	if err != nil {
+		return nil, err
+	}
 	cp.XMLNS = Strptr(CENC_XMLNS)
+	prSystemID, err := hex.DecodeString(CONTENT_PROTECTION_PLAYREADY_SCHEME_HEX)
+	if err != nil {
+		panic(err.Error())
```

Comment:
So, I'm okay with this in this situation. However, as a general rule of thumb, avoid panics. I probably would not accept this code as is in a future PR.